cmd/crawler/node/node.go

```diff
@@ -283,10 +283,6 @@ func (node *Node) Context() context.Context {
 }
 
 func (node *Node) chainStateHandler(ctx context.Context, msg interface{}, stream libp2pcore.Stream) error {
-	defer func() {
```

Comment:
after doing the fix, where you release your stream (`closeSteam(stream)`),  when the context `cancle()` called, will the stream be released automatically?